File: src/main/java/net/thenamedev/legendarena/listeners/PlayerDamageListener.java
package net.thenamedev.legendarena.listeners;

import org.bukkit.GameMode;
import org.bukkit.entity.Player;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;

/**
 * Damage listener for instant respawns.
 *
 * @author ThePixelDev
 */
public class PlayerDamageListener implements Listener {

    public void onDamage(EntityDamageEvent ev) {
        if(!(ev.getEntity() instanceof Player))
            return;
        Player p = (Player) ev.getEntity();
        if(p.getHealth() == 0.0 || p.getHealth() - ev.getDamage() == 0.0) {
            ev.setCancelled(true);
            //p.teleport()
            p.setGameMode(GameMode.SPECTATOR); //temporary gamemode setter
        }
    }

}

File: src/old/main/java/legendarena/listeners/PlayerDamageListener.kt
/*
 * This plugin is copyright Legend Arena, 2014-current. See LICENSE.md for full license file.
 */

package legendarena.listeners

import legendarena.hub.HubWarper
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent

public class PlayerDamageListener : Listener {

    EventHandler
    public fun listenForDamage(ev: EntityDamageEvent) {
        if(ev.getEntity() !is Player) return
        var p = ev.getEntity() as Player
        if(HubWarper.isExempt(p.getUniqueId())) return
        ev.setCancelled(true)
    }

}